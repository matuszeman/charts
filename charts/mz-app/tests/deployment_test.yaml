suite: deployment
templates:
  - deployment.yaml
tests:
  - it: should match snapshot with values-ci.yaml
    values:
      - ../values-ci.yaml
    set:
      containers:
        app:
          imageRepository: private
    asserts:
      - matchSnapshot: {}
  - it: public image repo
    values:
      - values/minimum.yaml
    set:
      containers:
        app:
          image: public-image
          imageRepository: public
    asserts:
      - equal:
          path: spec.template.spec.containers[0].image
          value: public.ecr.aws/abcd1245/public-image:0.1.0
  - it: dockerhub image repo
    values:
      - values/minimum.yaml
    set:
      containers:
        app:
          image: username/image
          imageRepository: dockerhub
    asserts:
      - equal:
          path: spec.template.spec.containers[0].image
          value: docker.io/username/image:0.1.0
