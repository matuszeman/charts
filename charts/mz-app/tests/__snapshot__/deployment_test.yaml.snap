should match snapshot with values-ci.yaml:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: app-name
        app.kubernetes.io/version: 0.1.0
        helm.sh/chart: mz-app-0.1.0
      name: RELEASE-NAME-resource-suffix
    spec:
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: app-name
      template:
        metadata:
          labels:
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: app-name
        spec:
          affinity:
            podAffinity:
              requiredDuringSchedulingIgnoredDuringExecution:
                - labelSelector:
                    matchExpressions:
                      - key: security
                        operator: In
                        values:
                          - S1
                  topologyKey: topology.kubernetes.io/zone
            podAntiAffinity:
              preferredDuringSchedulingIgnoredDuringExecution:
                - podAffinityTerm:
                    labelSelector:
                      matchExpressions:
                        - key: security
                          operator: In
                          values:
                            - S2
                    topologyKey: topology.kubernetes.io/zone
                  weight: 100
          containers:
            - image: 1234567890.dkr.ecr.eu-central-1.amazonaws.com/idp-example-web-app-apache:0.1.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /
                  port: apache
              name: apache
              ports:
                - containerPort: 80
                  name: apache
                  protocol: TCP
              readinessProbe:
                httpGet:
                  path: /
                  port: apache
              resources:
                limits:
                  cpu: 200m
                  memory: 256Mi
                requests:
                  cpu: 100m
                  memory: 128Mi
            - env:
                - name: configFromAwsSecret
                  valueFrom:
                    secretKeyRef:
                      key: somekey
                      name: RELEASE-NAME-resource-suffix-config-app-secrets
                - name: configFromConfigContent
                  valueFrom:
                    configMapKeyRef:
                      key: plaintextkey
                      name: RELEASE-NAME-resource-suffix-config-app
                - name: configFromConfigMap
                  valueFrom:
                    configMapKeyRef:
                      key: somekey
                      name: RELEASE-NAME-not-managed-cm
                - name: configFromSecret
                  valueFrom:
                    secretKeyRef:
                      key: somekey
                      name: RELEASE-NAME-not-managed-secret
                - name: configKeyTemplatedValue
                  valueFrom:
                    configMapKeyRef:
                      key: cm-key
                      name: app-config-RELEASE-NAME
                - name: configKeyValue
                  valueFrom:
                    configMapKeyRef:
                      key: cm-key
                      name: app-config
                - name: floatValue
                  value: "456.789"
                - name: intValue
                  value: "123"
                - name: secretKeyTemplatedValue
                  valueFrom:
                    secretKeyRef:
                      key: secret-key
                      name: app-secrets-RELEASE-NAME
                - name: secretKeyValue
                  valueFrom:
                    secretKeyRef:
                      key: some-key
                      name: app-secrets
                - name: stringValue
                  value: string
                - name: templatedValue
                  value: value-RELEASE-NAME
              image: 1234567890.dkr.ecr.eu-central-1.amazonaws.com/test-server:0.1.0
              imagePullPolicy: IfNotPresent
              livenessProbe:
                httpGet:
                  path: /
                  port: app
              name: app
              ports:
                - containerPort: 9000
                  name: app
                  protocol: TCP
                - containerPort: 9000
                  name: anotherport
              readinessProbe:
                httpGet:
                  path: /
                  port: app
              resources:
                limits:
                  cpu: 200m
                  memory: 256Mi
                requests:
                  cpu: 100m
                  memory: 128Mi
              startupProbe:
                failureThreshold: 30
                httpGet:
                  path: /
                  port: app
                periodSeconds: 10
              volumeMounts:
                - mountPath: /app-config
                  name: config-app
                  readOnly: true
                - mountPath: /app-secrets
                  name: config-app-secrets
                  readOnly: true
                - mountPath: /tmp/example.txt
                  name: config-not-managed-cm
                  readOnly: true
                  subPath: example.txt
                - mountPath: /etc/php.ini
                  name: config-php
                  readOnly: true
                  subPath: php.ini
                - mountPath: /app/public/var
                  name: ephemeral-app-cache
                  readOnly: false
          securityContext:
            sysctls:
              - name: net.ipv4.ip_unprivileged_port_start
                value: "0"
          serviceAccountName: RELEASE-NAME-resource-suffix
          volumes:
            - configMap:
                name: RELEASE-NAME-resource-suffix-config-app
              name: config-app
            - name: config-app-secrets
              secret:
                secretName: RELEASE-NAME-resource-suffix-config-app-secrets
            - configMap:
                name: RELEASE-NAME-not-managed-cm
              name: config-not-managed-cm
            - name: config-not-managed-secret
              secret:
                secretName: RELEASE-NAME-not-managed-secret
            - configMap:
                name: RELEASE-NAME-resource-suffix-config-php
              name: config-php
            - emptyDir:
                sizeLimit: 10Mi
              name: ephemeral-app-cache
